### PR TITLE
[vcpkg] Fix native platform expression

### PR DESCRIPTION
**Pull request by strega-nil**

### CODE REVIEW

File: azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1 [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    $host_triplet = 'x64-windows'
+} else {
+    Write-Warning "Unknown platform."
+    return
+}
+
+
+Write-Trace "test native qualified dependencies"
+
+$Overlays = "--overlay-ports=$PSScriptRoot/../e2e_ports/overlays", "--overlay-triplets=$PSScriptRoot/../e2e_ports/triplets"
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1:20]: This is already part of the common args; we should just use those: https://github.com/microsoft/vcpkg-tool/blob/7f762bbaece17fe644e166b90e2cdacd429f55c9/azure-pipelines/end-to-end-tests-prelude.ps1#L15

File: azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1 [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Run-Vcpkg x-set-installed
+Throw-IfFailed
+Run-Vcpkg install @Overlays --triplet $Triplet --host-triplet $Triplet vcpkg-native-dependency
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1:24]:
```suggestion
Run-Vcpkg install @commonArgs --host-triplet $Triplet vcpkg-native-dependency:$Triplet
```

File: azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1 [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+Throw-IfFailed
+$ports = Run-Vcpkg list | % { (-split $_)[0] } | % { ($_ -split ':')[0] }
+
+if ('vcpkg-empty-port' -notin $ports) {
+    throw "native platform dependency incorrect - vcpkg-empty-port not installed"
+}
+
+Run-Vcpkg x-set-installed
+Throw-IfFailed
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1:33]:
```suggestion
Refresh-TestRoot
```

File: azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1 [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+Run-Vcpkg install @Overlays --triplet $Triplet --host-triplet $Triplet vcpkg-native-dependency
+Throw-IfFailed
+$ports = Run-Vcpkg list | % { (-split $_)[0] } | % { ($_ -split ':')[0] }
+
+if ('vcpkg-empty-port' -notin $ports) {
+    throw "native platform dependency incorrect - vcpkg-empty-port not installed"
+}
```
Comment (ras0219-msft) [inline azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1:30]: We typically test installation of packages via
```powershell
Require-FileExists "$installRoot/$Triplet/share/vcpkg-empty-port"
```

File: azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1 [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+if ($IsLinux) {
+    $triplet = 'x64-linux-e2e'
+    $host_triplet = 'x64-linux'
+} else if ($IsMacOS) {
+    $triplet = 'x64-osx-e2e'
+    $host_triplet = 'x64-osx'
+} else if ($IsWindows) {
+    $triplet = 'x64-windows-e2e'
+    $host_triplet = 'x64-windows'
```
Comment (BillyONeal) [inline azure-pipelines/end-to-end-tests-dir/native-platform-dep.ps1:11]: I believe we default this to x86-windows?

**Review by BillyONeal — APPROVED** [review]

**Comment by autoantwort** [comment]
Without this the vcpkg ci command currently crashes

**Pull request merged**